cmake/extensions.cmake

```diff
@@ -2261,3 +2263,368 @@ function(target_byproducts)
                      COMMENT "Logical command for additional byproducts on target: ${TB_TARGET}"
   )
 endfunction()
+
+########################################################
+# 4. Zephyr devicetree function
+########################################################
+# 4.1. dt_*
+#
+# The following methods are for retrieving devicetree information in CMake.
+#
+# Note: In CMake we refer to the nodes using the node's path, therefore there
+# is no dt_path(...) function for obtaining a node identifier.
+
+# Usage:
+#   dt_nodelabel(<var> LABEL <label>)
+#
+# function for retrieving the node path for the node having nodelabel
+# <label>.
+#
+# the node path will be returned in the <var> parameter.
+#
+# <var>          : Return variable where the node path will be stored
+# LABEL <label>  : Node label
+function(dt_nodelabel var)
+  set(req_single_args "LABEL")
+  cmake_parse_arguments(DT_LABEL "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_nodelabel(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_LABEL_${arg})
+      message(FATAL_ERROR "dt_nodelabel(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  get_target_property(${var} devicetree_target "DT_LABEL|${DT_LABEL_LABEL}")
+  if(${${var}} STREQUAL ${var}-NOTFOUND)
+    set(${var})
+  endif()
+
+  set(${var} ${${var}} PARENT_SCOPE)
+endfunction()
+
+# Usage:
+#   dt_node_exists(<var> NODE <node>)
+#
+# Does a node path refer to a node?
+#
+# Tests whether a node path refers to a node which exists, i.e. is defined
+# in the devicetree.
+#
+# the result of the check will be returned in the <var> parameter.
```

Comment:
What is returned in <var> if the node exists, what is returned if the node doesn't exist

Reply:
Every node in the system will have the `DT_NODE|<node-path>` value TRUE, see https://github.com/zephyrproject-rtos/zephyr/blob/a3400ceba261351b435174f8980b9447e3dfea09/scripts/dts/dts_cmake.py#L42
meaning that this line will return TRUE if the node exists:
https://github.com/zephyrproject-rtos/zephyr/blob/a3400ceba261351b435174f8980b9447e3dfea09/cmake/extensions.cmake#L2342
but if the node did not exist, then `FALSE` is returned here:
https://github.com/zephyrproject-rtos/zephyr/blob/a3400ceba261351b435174f8980b9447e3dfea09/cmake/extensions.cmake#L2344

So answer is:
- TRUE when the node exists
- FALSE otherwise

Reply:
> So answer is:

I think @galak is asking you to document the behavior

Reply:
I felt he was asking also because he was unsure, but anything that is not clear in the description of the function itself should of course also be improved.

I just wanted to give a detailed explanation here first, before updating the PR in general with all requested changes.

Reply:
Function description improved